nft_ingester/src/bin/slot_persister/main.rs

```diff
@@ -497,3 +559,261 @@ async fn process_slots(
+        let higher_slot_boundary = get_last_persisted_slot(target_db.clone())
+            .saturating_sub(MISSED_SLOT_COLLECTION_OFFSET);
+        let slot_backfill_offset =
+            higher_slot_boundary.saturating_sub(lower_slot_boundary).min(slot_backfill_offset);
+        let mut last_valid_key = lower_slot_boundary;
+        info!(%lower_slot_boundary, "Acquiring missed slots starting at last checkpoint");
+
+        // Store missing slots
+        let missing_slots = Arc::new(Mutex::new(Vec::new()));
+
+        info!(
+            "Starting to collect slots from {} to {}",
+            lower_slot_boundary,
+            lower_slot_boundary + slot_backfill_offset
+        );
+
+        // Collect slots from last persisted slot down to 0
+        in_mem_dumper.clear().await;
+
+        // Start slot collection
+        let _ = slots_collector
+            .collect_slots(
+                &BUBBLEGUM_PROGRAM_ID,
+                lower_slot_boundary + slot_backfill_offset,
+                lower_slot_boundary,
+                cancellation_token.child_token(),
+            )
+            .await;
+
+        // Get the collected slots
+        let mut collected_slots = in_mem_dumper.get_sorted_keys().await;
+        in_mem_dumper.clear().await;
+        collected_slots.retain(|s| *s > lower_slot_boundary);
+
+        if cancellation_token.is_cancelled() {
+            info!("Shutdown signal received, stopping...");
+            return None;
+        }
+
+        let total_slots_to_check = collected_slots.len() as u64;
+
+        info!(
+            "Collected {} slots in range {} to {}",
+            total_slots_to_check,
+            lower_slot_boundary,
+            lower_slot_boundary + slot_backfill_offset
+        );
+
+        if collected_slots.is_empty() {
+            return Some((
+                lower_slot_boundary,
+                lower_slot_boundary + slot_backfill_offset,
+                Vec::new(),
+            ));
+        }
+
+        // Prepare iterators for collected slots and RocksDB keys
+        let mut slots_iter = collected_slots.into_iter();
+        let mut next_slot = slots_iter.next();
+
+        let cf_handle = target_db.db.cf_handle(RawBlock::NAME).unwrap();
+        let mut db_iter = target_db.db.raw_iterator_cf(&cf_handle);
+        db_iter.seek(lower_slot_boundary.to_be_bytes());
+
+        let mut current_db_slot = if db_iter.valid() {
+            if let Some(key_bytes) = db_iter.key() {
+                RawBlock::decode_key(key_bytes.to_vec()).ok()
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
+        // Verification loop
+        while let Some(slot) = next_slot {
+            if cancellation_token.is_cancelled() {
+                info!("Shutdown signal received, stopping...");
+                return None;
+            }
+
+            if let Some(db_slot) = current_db_slot {
+                if slot < lower_slot_boundary {
```

Comment:
this check doesn't look right to me. we're comparing our received slot with the lower bound (which should never be possible due to `collected_slots.retain(|s| *s > lower_slot_boundary);` above 
Did you mean to compare the db key? which also should not happen

Reply:
i agree, this was an initial check, but later i added the `retain`. will remove